app/clinic/controllers/labRequestController.js

```diff
@@ -0,0 +1,165 @@
+    .controller('LabRequestController', LabRequestController);
+
+  LabRequestController.$inject = ['$rootScope', '$stateParams', 'providerService', 'testProfileService', 'testService',
+    'notifier', '$filter', 'testOrderService', 'sessionService'];
+
+  function LabRequestController($rootScope, $stateParams, providerService, testProfileService, testService, notifier,
+    $filter, testOrderService, sessionService) {
+
+    var patientUuid = $stateParams.patientUuid;
+    var patient = {};
+    var vm = this;
+    vm.showMessages = false;
+
+    //requisições externas possuem alguns campos adicionais
+    vm.externalRequest = true;
+
+    vm.providers = [];
+    vm.profiles = [];
+    vm.tests = [];
+    vm.selectedTests = [];
+    vm.existingTestOrders = [];
+    vm.testsOfSelectedRequest = [];
+    vm.selectedProvider = null;
+    vm.date = null;
+    vm.selectedProfile = null;
+    vm.selectedTest = null;
+    vm.testOrderInDetail = null;
+
+    vm.addTest = addTest;
+    vm.addTestProfile = addTestProfile;
+    vm.removeTest = removeTest;
+    vm.saveTestOrderRequest = saveTestOrderRequest;
+    vm.showTestOrderDetails = showTestOrderDetails;
+    vm.deleteTest = deleteTest;
+    vm.resetForm = resetForm;
+
+    activate();
+
+    function activate() {
+      providerService.getProviders().then(function (providers) {
+        vm.providers = providers;
+      });
+      testProfileService.getTestProfiles().then(function (testProfiles) {
+        vm.profiles = testProfiles;
+      });
+      testService.getTests().then(function (tests) {
+        vm.tests = tests;
+      });
+      loadExistingTestOrders();
+    }
+
+    function loadExistingTestOrders() {
+      testOrderService.getTestOrdersByPatientUuid(patientUuid).then(function (testOrders) {
+        vm.existingTestOrders = testOrders;
+      });
+    }
+
+    function addTest() {
+      if (vm.selectedTest && vm.selectedTest.display) {
+        var alreadyContainsTest = vm.selectedTests.indexOf(vm.selectedTest) != -1;
+        if (!alreadyContainsTest) {
+          vm.selectedTest.profileName = null;
+          vm.selectedTests.push(vm.selectedTest);
+          vm.selectedTest = null;
+        } else {
+          notifier.error($filter('translate')('TEST_ALREADY_ADDED'));
+        }
+      } else {
+        notifier.error($filter('translate')('SELECT_TEST_FROM_LIST'));
+      }
+    }
+
+    function addTestProfile() {
+      if (vm.selectedProfile && vm.selectedProfile.name) {
+        var testUuids = vm.selectedProfile.tests;
+        testUuids.forEach(function (uuid) {
+          var test = getTestByUuid(uuid);
+          if (test != null) {
+            var alreadyContainsTest = vm.selectedTests.indexOf(test) != -1;
+            if (!alreadyContainsTest) {
+              test.profileName = vm.selectedProfile.name;
+              vm.selectedTests.push(test);
+            }
+          } else {
+            console.error("Teste com uuid " + uuid + " não encontrado");
```

Comment:
Tried to add the profile `TESTAGEM VIROLOGIA` and found the following error:
![image](https://user-images.githubusercontent.com/3810265/37647999-693974ce-2c36-11e8-8031-8609da3e194d.png)
However one test was added to the list:
![image](https://user-images.githubusercontent.com/3810265/37648046-8a3607dc-2c36-11e8-8442-c9554d79438c.png)

Reply:
The same for profile `BIOQUIMICA LABORATORIO`
![image](https://user-images.githubusercontent.com/3810265/37648157-d9fda7c0-2c36-11e8-8a19-7f8189b8ef91.png)

Reply:
And this after save:
![image](https://user-images.githubusercontent.com/3810265/37648414-919edfe8-2c37-11e8-98a3-418d4f37fb7f.png)

Reply:
And button `Details` doesn't work:
![image](https://user-images.githubusercontent.com/3810265/37648500-d221e844-2c37-11e8-94f8-f969fd59f5df.png)

Reply:
Some of the profiles are missing a few tests. Namely: `RPR, CD4%, CD4#, Baciloscopia, Triglicerides (TG)`. I think that the errors above are in the source of the problem.
![image](https://user-images.githubusercontent.com/3810265/37698462-c6b4166c-2ceb-11e8-8f83-a16a6c68a693.png)